### DISCUSSION TITLE
External File Upload and Download (like PDF, CSV, ...)

### DISCUSSION THREAD

**Discussion by trueberryless** [withastro/roadmap#732] [Proposal]
### Body

# Summary

Integrate the possibility to have static files like PDFs, CSVs, Word, Excel, TXT, and so on in the working directory, which can then be accessed on the web page. This feature is specifically for Starlight - I would say.

# Background & Motivation

It would be very nice if I can make files available on my Starlight website. For example, users who read my documentation want to be provided with files which the regularly have to use and don't want to search anywhere on the internet.

The functionality to store static files in the working directory where Astro / Starlight runs (in starlight the file can exists near the markdown docs) and then include an Astro component which shows a download button would be absolutely fantastic.

# Goals

A **concise, bulleted-list** outlining the intended goals of this RFC. 

- Detect static files
- Make them available to users by a download button
- Make them available to users by showing them on the website

# Example

This could be some inline button or a component which takes up the entire with and look even cleaner than this example:

![image](https://github.com/withastro/roadmap/assets/99918022/a08a3314-fc53-4841-9d71-680011f95942)

**Comment by delucis** [comment]
Hey @trueberryless! Good news — this should be possible already!

Any file that you place in the `public/` directory will automatically be copied over to your site’s build, allowing you to link to them, for example with a file structure like:

   ```
   public/
     pdfs/
       atmega.pdf
   src/
     content/
       docs/
         index.md
   ```
   
   Your `index.md` page could link to the PDF:
   
   ```md
   [ATmega328P & EEPROM](/pdfs/atmega.pdf)
   ```
   
   You could also force the link to download using HTML with the [`download` attribute](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/download):
   
   ```html
   <a href="/pdfs/atmega.pdf" download>ATmega328P & EEPROM</a>
   ```

**Reply by trueberryless** [reply]
WOW! This is amazing. Neither did I know that this is possible nor did I try to put the PDF in the public folder! Thank you so much for you quick response and anyone who puts work into these projects!
